torch/utils/_foreach_utils.py

```diff
@@ -5,7 +5,34 @@
 from torch import Tensor
 from torch.autograd.grad_mode import no_grad
 
+__all__ = ["SupportForeachDeviceType"]
 
+class SupportForeachDeviceType(object):
```

Comment:
Why not
```suggestion
class SupportForeachDevices(object):
```

I understand the idea of storing devices that support foreach, but I _don't_ understand why we have a _default_device_type attribute that we replace.